protobuf_serialization/reader.nim

```diff
@@ -164,12 +171,28 @@ proc readValueInternal[T: object](stream: InputStream, value: var T, silent: boo
             stream.readFieldPackedInto(fieldVar, header, ProtoType)
           else:
             stream.readFieldInto(fieldVar, header, ProtoType)
-        elif typeof(fieldVar) is ref and defined(ConformanceTest):
+        elif typeof(fieldVar) is ref and defined(npsConformanceTest):
           fieldVar = new typeof(fieldVar)
           stream.readFieldInto(fieldVar[], header, ProtoType)
         else:
           stream.readFieldInto(fieldVar, header, ProtoType)
 
+    # TODO preserve the unknown field
+    # maybe use the pragma proto2/3 to create a "hidden" unknownField
+    if not knownField:
+      if header.kind() == WireKind.Varint:
+        var flushed: uint64
+        stream.readFieldInto(flushed, header, puint64)
+      elif header.kind() == WireKind.Fixed64:
+        var flushed: uint64
+        stream.readFieldInto(flushed, header, fixed64)
+      elif header.kind() == WireKind.LengthDelim:
+        var flushed: seq[byte]
+        stream.readFieldInto(flushed, header, pbytes)
```

Comment:
ideally we should not have to actually copy the data here (and elsewhere)

Reply:
You mean that we should use `stream.advance` or something similar?

Reply:
yeah